deps/build.jl

```diff
@@ -121,60 +139,42 @@ if any(isfile.(hsl_archives))
     write(io, "\n")
   end
 
-  if isfile(hsl_ma57_archive)
-    @info "building ma57"
-    path_libhsl_ma57 = joinpath(libdir, "libhsl_ma57.$dlext")
-    open(path_deps, "a") do io
-      write(io, "const libhsl_ma57 = \"$path_libhsl_ma57\"\n")
-    end
-    if isfile(hsl_ma57_patch)
+  for software in keys(hsl_collection)
+    if haskey(hsl_archives, software) && isfile(hsl_archives[software])
+      @info "building $(software)"
+      path_libhsl = joinpath(libdir, "lib$(software).$dlext")
       open(path_deps, "a") do io
-        write(io, "const libhsl_ma57_patch = \"$hsl_ma57_patch\"\n")
+        write(io, "const lib$(software) = \"$(path_libhsl)\"\n")
       end
+      if haskey(hsl_patches, software) && isfile(hsl_patches[software])
+        open(path_deps, "a") do io
+          write(io, "const lib$(software)_patch = \"$(hsl_patches[software])\"\n")
```

Comment:
what's the lib_*_patch ?

Reply:
It was a path of a patch. It was generated by `BinaryProvider.jl`. The name was not user-friendly and it was not relevant to have a path if the software was patched.
I updated the code such that the string `lib_*_patch` was replaced by a boolean `*_patched`.